vcs/gitcmd/repo.go

```diff
@@ -150,23 +150,55 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-func (r *Repository) Branches() ([]*vcs.Branch, error) {
+// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
+func (r *Repository) branchCounts(branch, base string) (behind, ahead int, err error) {
+	cmd := exec.Command("git", "rev-list", "--count", "--left-right", base+"..."+branch)
```

Comment:
I agree.

To clarify, when I made the observation that it didn't seem to be possible to create branch names beginning with a dash, I did not mean to imply we could assume that's always the case. After all, maybe it's possible to create branches that begin with a dash via alternative means.

However, I thought there may be a way to make the branch name safe (even if it begins with a dash, etc.) by prepending "refs/heads/" to it.

But that doesn't seem to guarantee anything, so I will do it alongside with the `checkSpecArgSafety` check.